cache_test.go

```diff
@@ -225,7 +225,52 @@ func TestTTL(t *testing.T) {
 		t.Error("err should be nil", err.Error())
 	}
 	if ttl != 1 {
-		t.Fatalf("ttl should be 1, but %d return", ttl)
+		t.Fatalf("ttl should be 1, but %d returned", ttl)
+	}
+}
+
+func TestTouch(t *testing.T) {
+	cache := NewCache(1024)
+	key1 := []byte("abcd")
+	val1 := []byte("efgh")
+	key2 := []byte("ijkl")
+	val2 := []byte("mnop")
+	err := cache.Set(key1, val1, 1)
+	if err != nil {
+		t.Error("err should be nil", err.Error())
+	}
+	err = cache.Set(key2, val2, 1)
+	if err != nil {
+		t.Error("err should be nil", err.Error())
+	}
+	if touched := cache.TouchedCount(); touched != 0 {
+		t.Fatalf("touched count should be 0, but %d returned", touched)
+	}
+	err = cache.Touch(key1, 2)
+	if err != nil {
+		t.Error("err should be nil", err.Error())
+	}
+	time.Sleep(time.Second)
+	ttl, err := cache.TTL(key1)
+	if err != nil {
+		t.Error("err should be nil", err.Error())
+	}
+	if ttl != 1 {
+		t.Fatalf("ttl should be 1, but %d returned", ttl)
+	}
+	if touched := cache.TouchedCount(); touched != 1 {
+		t.Fatalf("touched count should be 1, but %d returned", touched)
+	}
+	val, err := cache.Get(key2)
+	if err == nil {
+		t.Fatal("key should be expired", string(val))
+	}
+	err = cache.Touch(key2, 2)
```

Comment:
If you Get key2 first, then the entry is already deleted, the logic of Touch expired key return not found is not covered.

Reply:
I have found that expired keys are still there if no evacuation has been done between a key's expiration and an attempt to access it. To get around this, `get()` specifically checks for expiration time and deletes the key, returning `ErrNotFound`:
https://github.com/coocood/freecache/blob/master/segment.go#L227
I had to do the same in `touch()`, otherwise it would resurrect an already expired key, which IMHO is counter-intuitive.
So this part of the test is to confirm `touch()` isn't extending TTL of keys that should no longer be there:
https://github.com/coocood/freecache/pull/79/files#diff-ba43b68f253651600e6bc26dc3758d1eR188

Reply:
I mean you can remove the line 264 to improve the test coverage.

Reply:
Oh, apologies for that, you're absolutely right, I forgot the `Get()` in there. Removed.